src/py/mat3ra/made/tools/build/interface/builders.py

```diff
@@ -284,3 +245,111 @@ def _update_material_name(
     ) -> Material:
         material.name = f"Twisted Nanoribbon Interface ({configuration.twist_angle:.2f}°)"
         return material
+
+
+class CommensurateLatticeInterfaceBuilderParameters(BaseModel):
+    """
+    Parameters for the commensurate lattice interface builder.
+
+    Args:
+        max_repetition_int (int): The maximum search range for commensurate lattices.
+        angle_tolerance (float): The tolerance for the angle between the commensurate lattice and the target angle.
+        return_first_match (bool): Whether to return the first match or all matches.
+    """
+
+    max_repetition_int: int = 10
+    angle_tolerance: float = 0.1
+    return_first_match: bool = False
+
+
+class CommensurateLatticeInterfaceBuilder(BaseBuilder):
+    _GeneratedItemType: type(CommensurateLatticePair) = CommensurateLatticePair  # type: ignore
+    _ConfigurationType: type(TwistedInterfaceConfiguration) = TwistedInterfaceConfiguration  # type: ignore
+
+    def _generate(self, configuration: _ConfigurationType) -> List[_GeneratedItemType]:
+        film = configuration.film
+        # substrate = configuration.substrate
+        max_search = self.build_parameters.max_repetition_int
+        a = film.lattice.vector_arrays[0][:2]
+        b = film.lattice.vector_arrays[1][:2]
```

Comment:
We should have a getter for this `film.lattice.a`, `film.lattice.b`, `film.lattice.c`

Reply:
Ok. We're adding this across all the made during cleanup task.